k8sutils/redis-cluster.go

```diff
@@ -171,18 +171,18 @@ func (service RedisClusterService) CreateRedisClusterService(cr *redisv1beta1.Re
 	serviceName := cr.ObjectMeta.Name + "-" + service.RedisServiceRole
 	logger := serviceLogger(cr.Namespace, serviceName)
 	labels := getRedisLabels(serviceName, "cluster", service.RedisServiceRole, cr.ObjectMeta.Labels)
-	annotations := generateServiceAnots(cr.ObjectMeta)
+	annotations := generateServiceAnots(cr.ObjectMeta, cr.Spec.KubernetesConfig.Service)
 	if cr.Spec.RedisExporter != nil && cr.Spec.RedisExporter.Enabled {
 		enableMetrics = true
 	}
 	objectMetaInfo := generateObjectMetaInformation(serviceName, cr.Namespace, labels, annotations)
 	headlessObjectMetaInfo := generateObjectMetaInformation(serviceName+"-headless", cr.Namespace, labels, annotations)
-	err := CreateOrUpdateService(cr.Namespace, headlessObjectMetaInfo, redisClusterAsOwner(cr), false, true)
+	err := CreateOrUpdateService(cr.Namespace, headlessObjectMetaInfo, redisClusterAsOwner(cr), false, true, cr.Spec.KubernetesConfig.Service.ServiceType)
```

Comment:
One more suggestion, I think we should not overwrite the existing headless service because this might be useful for internal communication without load-balancing. I would suggest creating another service in parallel with the defined service type and let's call it `additional-service.`  In this way, we can also remove the service creation logic from the helm chart.